sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientProvider.java

```diff
@@ -6,16 +6,22 @@
 import com.azure.core.http.HttpClient;
 import com.azure.core.http.HttpClientProvider;
 
+import java.util.concurrent.atomic.AtomicReference;
+
 /**
  * An {@link HttpClientProvider} that provides an implementation of HttpClient based on native JDK HttpClient.
  * <p>
  * NOTE: This implementation is only available in Java 11+ as that is when {@link java.net.http.HttpClient} was
  * introduced.
  */
 public final class JdkHttpClientProvider implements HttpClientProvider {
+    private static final AtomicReference<HttpClient> DEFAULT_HTTP_CLIENT = new AtomicReference<>();
 
     @Override
     public HttpClient createInstance() {
-        return new JdkAsyncHttpClientBuilder().build();
+        // by default use a singleton instance of http client
+        DEFAULT_HTTP_CLIENT.compareAndSet(null, new JdkAsyncHttpClientBuilder().build());
```

Comment:
here and similar places - could we avoid new XBuilder.build() if this method is called multiple times and some httpclient is already cached?

Reply:
@kasobol-msft 
With the current modification, the `DEFAULT_HTTP_CLIENT` is a shared atomic reference in objects created by JdkHttpClientProvider class. If the same JdkHttpClientProvider object calls the `createInstance` method multiple times.  `.compareAndSet` will return true for first time and then will return false because `DEFAULT_HTTP_CLIENT` no longer a null value reference. So `new JdkAsyncHttpClientBuilder().build()` will only be called at the first time.

Reply:
This isn't correct Shawn - we expect that the `compareAndSet` arguments are evaluated by the JVM before the call to `compareAndSet` happens, and so therefore every time it is called, we have first created a new `JdkAsyncHttpClientBuilder` and we've called `build()` on it. Kamil's concerns here are valid.